test/jdk/java/util/Collections/CheckedMapBash.java

```diff
@@ -183,4 +183,21 @@ public static Collection<Object[]> makeCheckedMaps() {
         };
         return Arrays.asList(params);
     }
+
+    @Test(groups = "type_check")
+    public static void testTypeCheck() {
+        try {
+            Map m = Collections.checkedMap(new HashMap<>(), Integer.class, Integer.class);
+            m.merge("key", "value", (v1, v2) -> null);
+            fail("Should throw ClassCastException");
+        } catch (ClassCastException ignore) {
```

Comment:
Since this test is already running with testNG, we should just use its builtin functionalities:
```java
    @Test(groups = "type_check")
    public static void testTypeCheck() {
        Map m = Collections.checkedMap(new HashMap<>(), Integer.class, Integer.class);
        Assert.assertThrows(ClassCastException.class, () -> m.merge("key", "value", (v1, v2) -> null));
        Assert.assertThrows(ClassCastException.class, () -> m.merge("key", 3, (v1, v2) -> v2));
    }
```

In addition, the checked map tests are in general, only checking the map functionality works without checking if the type checking functionality works. But adding a comprehensive test for checking functionalites would be much more work and is out of this patch's scope, so you don't need to worry about it.

Reply:
Thank you very much for informing me about these things. I have modified the test function name.